src/server/main_service.cc

```diff
@@ -539,6 +539,48 @@ ExecEvalState DetermineEvalPresense(const std::vector<StoredCmd>& body) {
   return ExecEvalState::SOME;
 }
 
+// Returns nullopt if the EXEC transaction is malformed i.e. has incompatible EVAL statements
+// Otherwise returns the mult mode for that transaction. Returns NOT_DETERMINED if no scheduling
+// is required.
+optional<Transaction::MultiMode> DeduceExecMode(ExecEvalState state,
+                                                const ConnectionState::ExecInfo& exec_info,
+                                                const ScriptMgr& script_mgr) {
+  // Check if script most LIKELY has global eval transactions
+  bool global_script = false;
+  Transaction::MultiMode multi_mode =
+      static_cast<Transaction::MultiMode>(absl::GetFlag(FLAGS_multi_exec_mode));
+
+  if (state != ExecEvalState::NONE) {
+    global_script = script_mgr.AreGlobalByDefault();
+
+    // Allow multi eval only when scripts run global and multi runs in global or lock ahead
+    // We adjust the atomicity level of multi transaction inside StartMultiExec i.e if multi mode is
+    // lock ahead and we run global script in the transaction then multi mode will be global.
+    if (!global_script || (multi_mode == Transaction::NON_ATOMIC)) {
```

Comment:
@chakaz FYI you will need to fix this in your code, as you run bullmq with lua lock ahead